sklearn/tests/test_common.py

```diff
@@ -163,6 +163,10 @@ def test_non_transformer_estimators_n_iter():
                              'LogisticRegressionCV']):
                     continue
 
+                # These models perform iterations for predict
+                if name in (['GaussianProcessClassifier']):
+                    continue
```

Comment:
max_iter is used during `predict()` not during `fit()` in `GaussianProcessClassifier`. The test checks that n_iter_ is set after fit(), which does not apply here.